enhancements/network/router-ca-configmap.md

```diff
@@ -0,0 +1,219 @@
+[NE-139](https://jira.coreos.com/browse/NE-139) describes the initial
+implementation.
+Following are the most salient PRs in the feature's history:
+
+* [openshift/cluster-ingress-operator#109 "Use self-signed default router certificate"](https://github.com/openshift/cluster-ingress-operator/pull/109)
+  added generation of default certificates and introduced the `router-ca`
+  ConfigMap.
+* [openshift/cluster-kube-apiserver-operator#222 "Trust cluster-ingress-operator's CA certificate"](https://github.com/openshift/cluster-kube-apiserver-operator/pull/222)
+  proposed adding `router-ca` to kube-apiserver's trust bundle; it was not
+  merged.  During review of this PR, it was determined that the ingress operator
+  must conditionally publish `router-ca`.
+* [openshift/cluster-kube-controller-manager-operator#145 "Trust cluster-ingress-operator's CA certificate"](https://github.com/openshift/cluster-kube-controller-manager-operator/pull/145)
+  added `router-ca` to kube-controller-manager's trust bundle.
+* [openshift/cluster-ingress-operator#110 "Put router-ca configmap in openshift-config-managed iff needed"](https://github.com/openshift/cluster-ingress-operator/pull/110)
+  changed the ConfigMap's namespace to `openshift-config-managed` and made its
+  publication conditional on its use.
+* [openshift/cluster-ingress-operator#142 "controller/certificate: relocate CA publishing"](https://github.com/openshift/cluster-ingress-operator/pull/142)
+  moved the `router-ca` publication to a separate controller.
+* [openshift/console-operator#328 "Bug 1764704: Sync router-ca to the console namespace"](https://github.com/openshift/console-operator/pull/328)
+  added `router-ca` to OpenShift Console's trust bundle.
+* [openshift/cluster-ingress-operator#329 "Always publish router-ca configmap"](https://github.com/openshift/cluster-ingress-operator/pull/329)
+  proposes making publication unconditional.
+
+## Drawbacks
+
+Publishing the ingress operator's signing certificate does not help
+if the administrator configures a custom default certificate.
+
+## Alternatives
+
+### Add a Cluster Configuration API for the Ingress CA
+
+Instead of using a ConfigMap, the ingress CA should be formal cluster
+configuration in the `spec` and `status` fields of the `default` cluster
+`ingresses.config.openshift.io` resource.  The `spec` field would be used to
+configure a custom ingress CA, and the `status` field would report the effective
+ingress CA.
+
+#### Drawbacks
+
+This approach would add a step to configuring a custom default certificate with
+custom PKI.
+
+The open questions around the semantics of multiple IngressControllers still
+need to be addressed with this approach.
+
+Network Edge has proposed this approach in the past and achieved no traction
+with it.
+
+### Inject the Ingress CA into ConfigMaps with a Well Known Label
+
+Instead of or in addition to publishing the `router-ca` ConfigMap, the ingress
```

Comment:
This sounds good to me.  Since it is `inject-ingress-cabundl` instead of `inject-default-cabundle`, will we be able to assume that we can simply use it? As a consumer, we don't have to care if it is default or custom?